src/cript/api/api.py

```diff
@@ -484,6 +502,404 @@ def _check_initial_host_connection(self) -> None:
+        host = Config.host
+        token = Config.api_token
+        print("---000----")
+        print(object_type.lower())
+        get_url = f"{host}/api/v1/{object_type.lower()}/{uuid}"
+        headers = {"Authorization": f"Bearer {token}", "Content-Type": "application/json"}
+        response = requests.get(get_url, headers=headers)
+        # print(response.json())
+        # print("are we fffing here")
+        response.raise_for_status()
+        if response.json()["data"]:
+
+            return response.json()["data"][0]  # Assuming the first item is the desired one
+
+        else:
+            raise ValueError(f"No data available in response {response}")
+
+    @staticmethod
+    def load_schema_for_node_type(node_type):
+        """
+        Load the validation schema for a given node type.
+
+        Args:
+            node_type (str): The type of the node (e.g., 'Project', 'Material').
+
+        Returns:
+            dict: The loaded schema.
+        """
+        schema_file_path = f"src/cript/schemas/{node_type.lower()}_schema.json"  # Path to the schema file
+        try:
+            with open(schema_file_path, "r") as file:
+                schema = json.load(file)
+            return schema
+        except FileNotFoundError:
+            print(f"Schema file not found for node type: {node_type}")
+
+        except json.JSONDecodeError:
+            print(f"Invalid JSON format in schema file for node type: {node_type}")
+
+    @staticmethod
+    def remove_keys_from_dict(
+        dictionary,
+        keys_to_remove=["uuid", "experiment_count", "inventory_count", "public", "created_at", "email", "model_version", "orcid", "uid", "updated_at", "username", "admin_count", "collection_count", "locked", "material_count", "member_count"],
+    ):
+        """Recursively remove keys from a dictionary."""
+        if not isinstance(dictionary, dict):
+            return dictionary
+        for key in keys_to_remove:
+            dictionary.pop(key, None)
+        for key, value in list(dictionary.items()):  # Use list() to avoid RuntimeError during iteration
+            if isinstance(value, dict):
+                API.remove_keys_from_dict(value, keys_to_remove)
+            elif isinstance(value, list):
+                for item in value:
+                    API.remove_keys_from_dict(item, keys_to_remove)
+        return dictionary
+
+    @staticmethod
+    def remove_duplicates(node_objects, duplicate_list):
+        removed_items = []  # To store removed materials
+        remaining_items = []  # To store remaining materials after removal
+
+        for item in node_objects:
+            if item["name"] in duplicate_list:
+                removed_items.append(item)
+            else:
+                remaining_items.append(item)
+
+        return remaining_items
+
+    @staticmethod
+    def send_patch_request_for_node(parent_node=None, payload=None):  # (self, base_url, entity_uuid, data, auth_token=None):
```

Comment:
this should not be a public function.
Users should never call this function.

Also, shouldn't be a static function, host, header etc. should be loaded from api class.

Reply:
I think here i will implement the capsule to resolve this